packages/puppeteer-core/src/cdp/Browser.ts

```diff
@@ -347,10 +347,19 @@ export class CdpBrowser extends BrowserBase {
     return await this.#defaultContext.newPage();
   }
 
-  async _createPageInContext(contextId?: string): Promise<Page> {
+  async _createPageInContext(
+    contextId?: string,
+    options?: CreatePageOptions,
+  ): Promise<Page> {
+    const hasTargets =
+      this.targets().filter(t => {
+        return t.browserContext().id === contextId;
+      }).length > 0;
     const {targetId} = await this.#connection.send('Target.createTarget', {
       url: 'about:blank',
       browserContextId: contextId || undefined,
+      // Works around buggy logic in CDP if newWindow is false.
+      newWindow: hasTargets && options?.type === 'window' ? true : undefined,
```

Comment:
Is there a crbug we can link to?

Reply:
no, I don't think anyone got time to fix it.